File: Data Structures & Algorithms/majority-element/submission-5.java
class Solution {
    private record MajorityHolder(int count, Integer el) {};

    public int majorityElement(int[] nums) {
        Map<Integer, Integer> counts = new HashMap<>();
        MajorityHolder maj = new MajorityHolder(Integer.MIN_VALUE, null);

        for (int n : nums)
        {
            int curr = counts.getOrDefault(n, 0) + 1;
            counts.put(n, curr);

            if (curr > maj.count())
            {
                maj = new MajorityHolder (curr, n);
            }
        }

        return maj.el();
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-5.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1

        while l <= r:
            m = l + (r-l) // 2
            curr = nums[m]

            if curr == target:
                return m

            if nums[l] <= curr:
                if target > curr or target < nums[l]:
                    l = m + 1
                else:
                    r = m - 1
            else:
                if target < curr or target > nums[r]:
                    r = m - 1
                else:
                    l = m + 1
            
        return -1